Client/Frontend/Home/TopSitesPanel.swift

```diff
@@ -167,12 +171,21 @@ class TopSitesPanel: UIViewController {
     }
 
     private func refreshTopSites(frecencyLimit: Int) {
+        // Don't allow Sync or other notifications to change the data source if we're deleting a thumbnail.
+        // If we are, we'll refresh at the end, so we can drop this one.
+        if !(self.collection?.userInteractionEnabled ?? true) {
```

Comment:
`refreshTopSites` is called as a result of notifications, and I believe it's consequently called on an arbitrary queue. You need to synchronously fetch this value on the main thread.